src/profile.ts

```diff
@@ -23,36 +25,72 @@ interface IProfilePki {
   publicKey: string;
 }
 
-export interface IProfile {
-  created: string;
-  updated: string;
+// Output from `everyauth profile export`
+interface IProfile {
+  profile: {
+    created: string;
+    updated: string;
+    account: string;
+    subscription: string;
+    baseUrl: string;
+    issuer: string;
+    subject: string;
+    keyPair: string;
+    kid: string;
+  };
+  type: string;
+  pki: IProfilePki;
+}
+
+// Output from `everyauth token`
+export interface IAuthedProfile {
   account: string;
   subscription: string;
   baseUrl: string;
-  issuer: string;
-  subject: string;
-  keyPair: string;
-  kid: string;
-  pki: IProfilePki;
-
-  token?: string;
+  accessToken: string;
+  expiresAt: string;
+  expiresAtMs: number;
 }
 
 export let cachedFoundProfile: IProfile;
-
-export const cachedJwt: {
-  expiresAt: number;
-  token: string;
-} = { expiresAt: 0, token: '' };
+export let cachedJwt: IAuthedProfile = {
+  account: '',
+  subscription: '',
+  baseUrl: '',
+  accessToken: '',
+  expiresAt: '',
+  expiresAtMs: 0,
+};
 
 /**
  * Supply a profile to use directly, instead of using the automatic discovery options.
  *
- * @param profile A profile object, created from a previously initialized profile.  Includes the private and
- * public keys used to authenticate with the remote service.
+ * @param profile A profile object, generated either via `everyauth profile export` or via `everyauth token`.  Can include the private and
+ * public keys used to authenticate with the remote service for ever-fresh credential generation.
  */
-export const config = async (profile: IProfile) => {
+export const config = async (profile: IProfile | IAuthedProfile) => {
+  if ('profile' in profile) {
+    cachedFoundProfile = profile;
+    return;
+  }
+
+  cachedJwt = profile;
+};
+
+export const loadEnvironmentToken = (): IAuthedProfile | undefined => {
+  const profile = process.env.EVERYAUTH_PROFILE_TOKEN
+    ? (cachedFoundProfile = JSON.parse(process.env.EVERYAUTH_PROFILE_TOKEN))
```

Comment:
Should this be base64-encoded? Setting a JSON to an env var in bash is going to be an escape hell.

Reply:
Went back and forth on this one.  Realized that all of the spaces likely to use this approach are automation-driven - CICD, etc - which handle that complexity, often via webforms.  If anything, would be amenable to adding a `_B64` variant to each of the options to offer that as a capability.

Reply:
+1 on at least having _B64 variant, at least lambda console does not support serializing JSON to an escaped format. also, JSONs can become a mess when accidentally console.log-ed in GHAs as it's not censored properly as we learned the hard way in the integrations repo a while ago. Are there any tradeoffs to make it all base64 anyways?

Reply:
B64 won't help console.log at all - in fact, it'll make it worse because it won't match the secret _at all_, whereas as long as it's not pretty-printed, it should still match what's in the environment and be censored.

Reply:
Base64 encoded string is way more user-friendly to pass around in the scripting context than pure JSON. B64 encoding and decoding in script is trivial in case anyone needs the JSON in script. I'd suggest we switch to B64, and if you see any use for plain JSON on top of that (I don't) maybe we can add an option for it as well?

Reply:
I support using base64, since is a recommended practice, especially for multiline values.